Firebase/Database/Api/FIRDatabase.m

```diff
@@ -38,142 +41,27 @@ @interface FIRDatabase ()
 
 @implementation FIRDatabase
 
-/** A NSMutableDictionary of FirebaseApp name and FRepoInfo to FirebaseDatabase instance. */
-typedef NSMutableDictionary<NSString *, NSMutableDictionary<FRepoInfo *, FIRDatabase *> *> FIRDatabaseDictionary;
-
 // The STR and STR_EXPAND macro allow a numeric version passed to he compiler driver
 // with a -D to be treated as a string instead of an invalid floating point value.
 #define STR(x) STR_EXPAND(x)
 #define STR_EXPAND(x) #x
 static const char *FIREBASE_SEMVER = (const char *)STR(FIRDatabase_VERSION);
 
-+ (void)load {
-  NSNotificationCenter *center = [NSNotificationCenter defaultCenter];
-  [center addObserverForName:kFIRAppDeleteNotification
-                      object:nil
-                       queue:nil
-                  usingBlock:^(NSNotification * _Nonnull note) {
-      NSString *appName = note.userInfo[kFIRAppNameKey];
-      if (appName == nil) { return; }
-
-      FIRDatabaseDictionary* instances = [self instances];
-      @synchronized (instances) {
-          NSMutableDictionary<FRepoInfo *, FIRDatabase *> *databaseInstances = instances[appName];
-          if (databaseInstances) {
-              // Clean up the deleted instance in an effort to remove any resources still in use.
-              // Note: Any leftover instances of this exact database will be invalid.
-              for (FIRDatabase * database in [databaseInstances allValues]) {
-                  [FRepoManager disposeRepos:database.config];
-              }
-              [instances removeObjectForKey:appName];
-          }
-      }
-  }];
-}
-
-/**
- * A static NSMutableDictionary of FirebaseApp name and FRepoInfo to
- * FirebaseDatabase instance. To ensure thread-safety, it should only be
- * accessed in databaseForApp:URL:, which is synchronized.
- *
- * TODO: This serves a duplicate purpose as RepoManager.  We should clean up.
- * TODO: We should maybe be conscious of leaks and make this a weak map or
- * similar but we have a lot of work to do to allow FirebaseDatabase/Repo etc.
- * to be GC'd.
- */
-+ (FIRDatabaseDictionary *)instances {
-    static dispatch_once_t pred = 0;
-    static FIRDatabaseDictionary *instances;
-    dispatch_once(&pred, ^{
-        instances = [NSMutableDictionary dictionary];
-    });
-    return instances;
-}
-
 + (FIRDatabase *)database {
-    if (![FIRApp isDefaultAppConfigured]) {
```

Comment:
This is still worth keeping since `[FIRApp defaultApp]` will log an error anyways and it's a developer error.

Reply:
Done